modules/ROOT/pages/osv2-tutorial.adoc

```diff
@@ -5,20 +5,339 @@ endif::[]
+the Object Store.
+
+Why is the value binary? Mule 4 wraps values into a Mule object that causes 
+them to be only visible in binary through Anypoint Platform. Behind the scenes, Mule 4 
+executes binary serialization with the Mule internal serializer. The user interface cannot
+deserialize the object, hence it can only tell you that the value is now binary in the user
+interface. 
+
+But you probably want clarification that your value worked too. For this we need to return to 
+Design Center and add a new flow that retrieves a value.
+
+=== Retrieve the Value
+
+. Click the hamburger icon and click *Design Center*.
+. In Design Center, click your project, and click *Open*.
+. Open the rectangular box icon to the left of Project to expand the Flow menu.
+. First rename your existing flow by clicking the vertical dot icon and clicking *Rename* from
+the menu. Change "New Flow" to "Store".
++
+image::os-dc-flows.png[]
++
+. Click the plus sign to the right of Flows.
+. Click *Go straight to canvas* to exit the wizard.
+. In the Trigger menu, choose *HTTP Listener*.
+. For Path, specify the `/retrieve` value.
++
+image:os-dc-http-retrieve.png[]
++
+Because the default settings for the HTTP Listener are correct, there's no need to click Edit. 
+However you can if you want to verify that the host is 0.0.0.0 and the port is 8081. These 
+are the same as in the previous Store flow.
++
+. Now that we have started the flow, you can now rename it to Retrieve.
+. Click the plus sign next to the HTTP Listener to add a component.
+. In *Select a component*, search for "Obj", and click the *ObjectStore Connector*.
+. Click the *Retrieve* operation and in the Key field, specify the `#[attributes.queryParams.key]` value.
+
+This completes the Retrieve flow. Click Run again to restart your application.
+
+In the command prompt, submit this curl command to view the key's value:
+
+[source,console]
+----
+curl http://osv2-qoav.us-w1.cloudhub.io/retrieve?key=MyKey;echo
+----
+
+(The `;echo` statement at the end of the command makes the output more readable.)
+
+The output from this command lists the value for the key:
+
+[source,console]
+----
+"OSv2 Test"
+----
+
+From this exercise, the Object Store user interface helps you
+see that the application you created in Design Center can be accessed using a curl command.
+
+For future use of Object Store in Design Center, you don't need to view the Runtime Manager interface each
+time you update the Object Store. After you create an application linked to the OSv2 connector, you can
+store a key and value in the Object Store with one curl command and view the results
+with another. Just change the key and/or the value as needed.
+
+Store:
+
+[source,console]
+----
+curl -X POST -H "Content-Type: application/json" -d '{ "key": "MyKey", "value": "OSv2 Test" }' "http://osv2-example.us-w1.cloudhub.io/store"
+----
+
+Retrieve:
+
+[source,console]
+----
+curl http://osv2-qoav.us-w1.cloudhub.io/retrieve?key=MyKey
```

Comment:
Same thing here:
Doesn't this link need to change based on the customer's application name?  Shouldn't we use the same approach as was used above where we make the user copy in the link from design center as LINK_FROM_DC in the url?

Reply:
Fixed. The curl commands now have either LINK_FROM_DESIGN_CENTER or LINK_FROM_RUNTIME_MANAGER depending on the section.